### PR TITLE
SYL-4137: Remove language flags

### PR DESCRIPTION
**Pull request by bartek-sek**

### CODE REVIEW

**Review by mpysiak — DISMISSED** [review]

File: src/Sylius/Bundle/AdminBundle/templates/product/form/sections/attributes/list/translations/flag.html.twig
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/templates/product/form/sections/attributes/list/translations/flag.html.twig:1]: Due to the BC promise, we cannot delete the existing templates, mark this one as deprecated

**Review by GSadee — COMMENTED** [review]
Beyond ADR, we should mention this change in the UPGRADE file

**Comment by bartek-sek** [comment]
> Beyond ADR, we should mention this change in the UPGRADE file

Done

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -2159,6 +2159,7 @@ Similarly, use form types from `ShopBundle` for the Shop context.
 * `use_webpack` twig global variable was removed. Webpack is always used now, and there is no need to check for it.
 * Image sizes have been simplified and standardized for both the Admin and Shop Bundle.
 * Some Twig extension services have been moved from the UiBundle to the new Twig Extra package
+* Flags have been removed from languages - it resolves all ambiguity and ensures a consistent and straightforward language selection experience. This approach reduces maintenance complexity and eliminates the risk of incorrect or misleading flag assignments.
```
Comment (GSadee) [inline UPGRADE-2.0.md:2162]: This note should be moved to the section with the upgrade from `2.0.0` to `2.0.1`

**Review by GSadee — APPROVED** [review]

**Review by mpysiak — APPROVED** [review]

**Pull request merged**